### DISCUSSION TITLE
Compilation errors (wrong version 61.0)

### DISCUSSION THREAD

**Discussion by gwkoch** [mjansen4857/pathplanner#171] [General]
I apologize if I'm not posting this in the right place, but I figured a discussion thread might be more useful than sending you a direct email.

Our team has been using PathPlanner and enjoying it throughout the last year and a half. Unfortunately, when updating to the latest version (installed via Windows Store and also added Pathplannerlib.json as a vendor library), we were unable to import a lot of the classes. Here is an example of the errors we are getting when we try to build our command-based project for our robot.

```
class file has wrong version 61.0, should be 55.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.

bad class file: C:\Users\gkoch\.gradle\caches\modules-2\files-2.1\com.pathplanner.lib\PathplannerLib-java\2023.1.0\afbe981ce8fa10f758b53ae8931024a08d073be\PathplannerLib-java-2023.1.0.jar(/com/pathplanner/lib/commands/PPSwerveControllerCommand.class)
```

My best guess is that there is a Java compatibility error, but even after updating my JRE version from 11 to 17,  updating the source and target compatibility fields in the build.gradle, and setting the JAVA_HOME environment variables, vscode gives:

```
What went wrong:
Execution failed for task ':compileJava'.
> error: invalid source release: 17

```

Am I on the wrong track here? Any idea what I need to do to get your libraries working? We greatly appreciate any insight.

**Comment by mjansen4857** [comment]
Are you using the 2022 or 2023 version of wpilib/vscode? It should work fine with the 2023 version. I believe I can make a change to the PPLib gradle config to build with java 11 compatibility. I will try to do that for the next release. No idea when that will be, but I will likely do one within the next week or 2.

**Comment by mjansen4857** [comment]
You can download the maven artifact from [here](https://github.com/mjansen4857/pathplanner/actions/runs/3570171473) and replace the files in your gradle cache and that will fix your issue.

**Comment by gwkoch** [comment]
We're on 2022 WPILib/vscode (https://github.com/wpilibsuite/allwpilib/releases/tag/v2022.4.1).  I wasn't aware that 2023 is even available (are you on the beta?).  Only 2022 downloads are offered from the WPILib installation page.

Thanks for the link to the maven artifact. Going to be honest though, I have no idea what you mean by replacing the files in the gradle cache. Never done that sort of thing. I assume I'm supposed to do something with all the .jar files in PPLib-Maven\maven\development\com\pathplanner\lib\PathplannerLib-java\2023.1.0 directory.

I did a bit of searching but the Gradle tutorials are pretty vast. Where do I go in our project to do this?

**Comment by mjansen4857** [comment]
From your error output above, your cache is at `C:\Users\gkoch\.gradle\caches\modules-2\files-2.1\com.pathplanner.lib\PathplannerLib-java`. Delete the 2023.1.0 directory from there and replace it with the one from the zip file.

EDIT: The setup of the cache at this location is different, you're probably better off just deleting the pathplanner folder here and using the one mentioned below.

And yes, the beta is available from the wpilib github repo.

**Reply by mjansen4857** [reply]
You may need to do the same thing in `C:\Users\Public\wpilib\2022\maven\com\pathplanner\lib\PathplannerLib-java` if it ends up using that one

**Reply by sciencewhiz** [reply]
> And yes, the beta is available from the wpilib github repo.

but someone without beta access wouldn't have access to Game Tools which has the image needed to run the beta version of wpilib.

**Comment by gwkoch** [comment]
Thanks for your prompt replies. What I found on my end is that your first suggestion didn't make any difference, so I just got rid of that directory as you suggested.

Is it a bad thing that the C:\Users\Public directory didn't have a pathplanner folder at all inside the maven\com?  I copy-pasted the pathplanner folder from the maven artifact, since there wasn't even any to begin with. The project now successfully built. Hopefully can try deploying this code on a robot tomorrow.

Follow-up question: If we used the 2023 version of WPILib/vscode, I assume we wouldn't have needed to do any of this?  It seems this occurred just due to a version conflict compared to your project's compilation that I could not easily resolve on my end alone.

**Reply by mjansen4857** [reply]
Not a bad thing, its just one of the possible places that it looks for the library. If you used the 2023 beta you could just use it normally, yes. But after the next PPLib update 2022 should just update and be fine as well.
